.changeset/empty-fans-reflect.md

```diff
@@ -0,0 +1,53 @@
+---
+'astro': minor
+---
+
+Adds a new `experimental.tsconfig` configuration option
+
+Enables a streamlined way of configuring TypeScript in your project. Enjoy an improved DX when setting up Astro for Intellisense, type-checking, and more.
+
+This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding your build from your checked files by default.
+
+To enable this option in an existing project, you will need to update your `tsconfig.json` to reference the new file. You may also need to update your `astro.config.mjs` as some items have now moved to Astro configuration options.
+
+Your `tsconfig.json` file must now extend the new `/.astro/` file:
+
+```diff
+{
+    "extends": [
+        "astro/tsconfigs/base", 
++        "./.astro/tsconfig.json"
+    ]
+}
+```
+
+If you have `include` or `exclude` inside your root `tsconfig.json`, move them to your Astro config:
+
+```diff
+{
+-    "include": ["src/**/*"],
+-    "exclude": ["node_modules"]
+}
+```
+
+```diff
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+    experimental: {
++        tsconfig: {
++            include: ['src/**/*'],
++            exclude: ['node_modules']
++        }
+    }
+})
+```
+
+It will create references under the hood, making `env.d.ts` optional if you haven't made any change to it:
```

Comment:
```suggestion
It will create references under the hood, making `env.d.ts` only necessary if you have added / for... :
```
Can you phrase it this way? What is this file still required for?

Reply:
Updated, not 100% about what to put. @Princesseuh wdyt?

Reply:
Sarah's suggestion is good for me!

Reply:
Good by me for docs! Sounds legit. :joy: Erika can say if it's the correct thing to say.